pkg/controller/notification/notification_controller.go

```diff
@@ -0,0 +1,143 @@
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// if is delivered, then check when status was changed and delete it if the requested duration has passed
+	completeCond, found := condition.FindConditionByType(notification.Status.Conditions, toolchainv1alpha1.NotificationSent)
+	if found && completeCond.Status == corev1.ConditionTrue {
+		deleted, requeueAfter, err := r.checkTransitionTimeAndDelete(reqLogger, notification, completeCond)
+		if deleted || err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: requeueAfter,
+		}, nil
+	}
+
+	reqLogger.Info("Notification has been delivered")
+	return reconcile.Result{
+		Requeue:      true,
+		RequeueAfter: r.config.GetDurationBeforeNotificationDeletion(),
+	}, r.setStatusNotificationDelivered(notification)
+}
+
+func (r *ReconcileNotification) checkTransitionTimeAndDelete(log logr.Logger, notification *toolchainv1alpha1.Notification, completeCond toolchainv1alpha1.Condition) (bool, time.Duration, error) {
+	log.Info("the Notification is delivered so we can deal with its deletion")
+	timeSinceCompletion := time.Since(completeCond.LastTransitionTime.Time)
+
+	if timeSinceCompletion >= r.config.GetDurationBeforeNotificationDeletion() {
+		log.Info("the Notification has been delivered for a longer time than the 'durationBeforeNotificationDeletion', so it's ready to be deleted",
+			"durationBeforeNotificationDeletion", r.config.GetDurationBeforeNotificationDeletion().String())
+		if err := r.client.Delete(context.TODO(), notification, &client.DeleteOptions{}); err != nil {
+			return false, 0, errs.Wrapf(err, "unable to delete Notification object '%s'", notification.Name)
+		}
+		return true, 0, nil
+	}
+	diff := r.config.GetDurationBeforeNotificationDeletion() - timeSinceCompletion
+	log.Info("the Notification has been completed for shorter time than 'durationBeforeNotificationDeletion', so it's going to be reconciled again",
+		"durationBeforeNotificationDeletion", r.config.GetDurationBeforeNotificationDeletion().String(), "reconcileAfter", diff.String())
+	return false, diff, nil
+}
+
+func (r *ReconcileNotification) updateStatusConditions(notification *toolchainv1alpha1.Notification, newConditions ...toolchainv1alpha1.Condition) error {
+	var updated bool
+	notification.Status.Conditions, updated = condition.AddOrUpdateStatusConditions(notification.Status.Conditions, newConditions...)
+	if !updated {
+		// Nothing changed
+		return nil
+	}
+	return r.client.Status().Update(context.TODO(), notification)
+}
+
+func (r *ReconcileNotification) setStatusNotificationDelivered(notification *toolchainv1alpha1.Notification) error {
```

Comment:
It says that notification has been (not) delivered everywhere in the code while it's actually about the notification being sent.

Reply:
@alexeykazakov yeah, i think Shane's PRs were using the term delivered but you had asked me to rename it to sent in api. However, there is code in host now that still uses the term "Deliver" or "Delivered" i'll change the code to use send and sent instead..